persist/sqlite/consensus_refactored_test.go

```diff
@@ -0,0 +1,275 @@
+package sqlite_test
+
+import (
+	"math"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"go.sia.tech/core/consensus"
+	"go.sia.tech/core/types"
+	"go.sia.tech/coreutils/chain"
+	ctestutil "go.sia.tech/coreutils/testutil"
+	"go.sia.tech/explored/explorer"
+	"go.sia.tech/explored/internal/testutil"
+	"go.sia.tech/explored/persist/sqlite"
+	"go.uber.org/zap/zaptest"
+)
+
+type node struct {
+	db    explorer.Store
+	store *chain.DBStore
+
+	blocks []types.Block
+	states []consensus.State
+}
+
+func newStoreRefactored(t *testing.T, v2 bool, f func(*consensus.Network, types.Block)) *node {
```

Comment:
I'd probably do either:

- `newTestChainV1`
- `newTestChainV2`

or (my preference I think)

`newTestChain(t *testing, network *consensus.Network, genesis types.Block)`
I would get rid of 'f', at the very least give it a more descriptive name.